test/CMakeLists.txt

```diff
@@ -123,6 +122,13 @@ if(NOT (MIOPEN_TEST_GFX900 OR MIOPEN_TEST_GFX906 OR MIOPEN_TEST_GFX908 OR MIOPEN
         set(MIOPEN_TEST_GPU_DETECTION_FAILED TRUE)
     endif()
 endif()
+
+set_var_to_condition(MIOPEN_LIBMLIR_SUPPORTS_GFX103X_DEFAULT ${MIOPEN_USE_MLIR} AND (${rocMLIR_VERSION_FLAT} GREATER 100000000))
+option( MIOPEN_LIBMLIR_SUPPORTS_GFX103X "libMLIR capability to support GFX10. Enables MLIR tests on GFX103X." ${MIOPEN_LIBMLIR_SUPPORTS_GFX103X_DEFAULT})
+
+set_var_to_condition(MIOPEN_TEST_MLIR_DEFAULT ${MIOPEN_USE_MLIR} AND NOT (${MIOPEN_TEST_GFX103X} AND NOT ${MIOPEN_LIBMLIR_SUPPORTS_GFX103X}))
+option( MIOPEN_TEST_MLIR "Test for MLIR compilation backend" ${MIOPEN_TEST_MLIR_DEFAULT} )
```

Comment:
@jerryyin 
> I want to point out that in the past, we don't maintain backward compatibility of the MIOpen tests...

Hmm... Good point. It was correct provided that MLIR is a part of MIOpen. Can you please [answer this Q](https://github.com/ROCmSoftwarePlatform/MIOpen/pull/1897#issuecomment-1360366088):

> ❓ **_Is MLIR library included into ROCm directly (as ROCm component) or it comes there only as a dependence of MIOpen?_**

Another question: do we want/require users of old ROCm to install the newest version of libMLIR? If not, then tests shall pass with vanilla old ROCm.

For now it looks like we want to maintain bwd compatibility (including tests).

Reply:
> ❓ Is MLIR library included into ROCm directly (as ROCm component) or it comes there only as a dependence of MIOpen?

MLIR library is not a ROCm component. It only ship as a dependence of MIOpen and MIGraphX. The only way a client like Pytorch can pull in MLIR is through MIOpen's `requirements.txt`.

> do we want/require users of old ROCm to install the newest version of libMLIR?

No one require us to give that guarantee. (and we don't test that scenario because of it.) Realistically speaking we haven't observed any version mismatch issues in the past too. Though most of the cases we don't have a dependency on ROCm version. ROCm update will only rarely break our internal test, there's a even smaller chance it break the static library.

Reply:
@jerryyin 
> MLIR library is not a ROCm component. It only ship as a dependence of MIOpen and MIGraphX. The only way a client like Pytorch can pull in MLIR is through MIOpen's requirements.txt.

If this is the case, then both #1913 and this PR can be closed.

Then we should require some actions from the end users who use old libMLIR but want to update MIOpen to the most recent version: _they must install the new version of libMLIR_ (or globally disable MLIR).

Please let me know your opinion.

Reply:
@atamazov Yes, I think enforcing them to always install the correct version of MLIR sounds like the right way to do it. Thinking about it, I think we can close #1913 and this PR but instead:
 - Bump rocMLIR version to 1.1.0 (and make sure that we bump at least the minor version in each ROCm release, like MIOpen do)
 - Amend https://github.com/ROCmSoftwarePlatform/MIOpen/blob/2d90671f257244fd0d17b686f4ac5c7ba4ebfbfd/CMakeLists.txt#L337 to `find_package(rocMLIR 1.0.0 EXACT CONFIG)`

Let me if you agree with it.

Reply:
@jerryyin On the other hand, it's good to have as much flexibility as possible. This may become handy in the future. For example, if the latest version of MIOpen can work with an outdated version of libMLIR, then we can use this possibility to triage issues.

Another imaginary example is when QA finds a complex bug in a release candidate branch, and that bug can be resolved by reverting to a previous version of libMLIR. Forcing MIOpen to always use the "correct" version of libMLIR would make this impossible.

Of course, maintaining flexibility takes effort, but it's often worth it. Your opinion?

Reply:
[Informative] @jerryyin 
> make sure that we bump at least the minor version in each ROCm release, like MIOpen do

We don't actually do this. Yes, we bump MIOpen version with each new release of _MIOpen_. And usually we release new MIOpen with each new ROCm release, but from time to time we include the same MIOpen (with the same version) into a new ROCm.

Reply:
@jerryyin 
> Bump rocMLIR version to 1.1.0

I agree 🟢

Reply:
@atamazov I'd say this is a good example on making tests backward compatible so let's ship it.

> Another imaginary example is when QA finds a complex bug in a release candidate branch, and that bug can be resolved by reverting to a previous version of libMLIR.

Realistically speaking this will probably never happen 😄. First off, QA doesn't find complex bug, they find the bare minimal to reproduce bug (usually build related). Secondly, they don't do any more triaging than what is necessary, and MLIR related issues usually ended up re-assigned to me. - end of the rant. Since those issues come back to me usually, I wasn't as considerate in MIOpen end as I should. But you are right, this can be convenient if in my own triage to test lastly released package and narrow the scope here.

Reply:
[Informative] @jerryyin I mean problems like the network non-convergence (or performance drops) that QA finds in MIOpen from time to time and assign to MIOpen.